src/pas/plugins/sqlalchemy/plugin.py

```diff
@@ -234,6 +234,17 @@ def group_class(self):
     # IUserManagement implementation
     #
 
+    security.declarePrivate('updateUser')
+    @graceful_recovery()
+    def updateUser(self, user_id, login_name):
+        session = Session()
+        principal = self._get_principal_by_id(user_id)
+        principal.zope_id = login_name
+        userquery = session.query(self.user_class).filter_by(zope_id=user_id)
+        userquery = userquery.first()
```

Comment:
This does not fail if no user can be found; perhaps you want to use the ``one()`` method? Otherwise, the method will fail in the next line with a "can't assign attribute to None object".